### ISSUE TITLE
Launching .excalidraw from file manager opens incorrect file in PWA

### ISSUE THREAD

**Issue by adam-wood** [excalidraw/excalidraw#5201]
When opening a *.excalidraw file from the windows file manager, the excalidraw PWA opens the wrong file. It launches PWA with the last open file active (not the launched file).

**Expected behaviour:** When opening a *.excalidraw file (eg by double click), the PWA should open and load that file.

### Steps to reproduce
Preconditions: PWA installed, file handling enabled.
1. Clear the canvas in excalidraw
2. Optional: Close all excalidraw  PWAs
3. Double click a saved *.excalidraw file (the file should have content)
4. PWA opens to blank canvas

### System
OS: Windows 11 22000.675
Browser: Edge 101.0.1210.47 
  File Handling API: Enabled
  File Handling Icons: Enabled

**Comment by jtpotato** [comment]
Having the same issue. I think it might be an Edge-specific thing as I was using the Chrome PWA a few days ago and it was fine.

**Comment by adam-wood** [comment]
Yes, confirming that the Chrome PWA works as expected.

**Comment by karlhorky** [comment]
I can confirm that double-clicking on an `.excalidraw` file on macOS also no longer opens the file.

Adding some context, as this `file_handlers` feature (associating `.excalidraw` files to open in the PWA) did work before, as part of the job to achieve parity with the deprecated Electron version:

https://blog.excalidraw.com/excalidraw-and-fugu/#:~:text=Now%20check%20out%20what%20happens%20when%20I%20right%2Dclick%20an%20Excalidraw%20file.%20I%20can%20choose%20to%20open%20it%20with%20Excalidraw%2C%20the%20installed%20PWA.%20Of%20course%20double%2Dclicking%20would%20work%2C%20too%2C%20it%E2%80%99s%20just%20less%20dramatic%20to%20demonstrate%20in%20a%20screencast.

It was originally a PR by @tomayac here (reviewed by @Lipis and @dwelle):

- https://github.com/excalidraw/excalidraw/pull/1736

**Comment by karlhorky** [comment]
In the meantime, on macOS, I'll try out this SwiftUI wrapper ExcalidrawZ by @chocoford:

- https://github.com/chocoford/ExcalidrawZ/

**Comment by tomayac** [comment]
I just tested this on macOS Sequoia (`15.1 Beta (24B5055e)`) with `'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36'` and it worked fine repeatedly. As a first step, can you uninstall the PWA and re-install it, so the file handler gets re-registered with the operating system? Then can you try with the trivial file below (download and rename it to `Untitled-2024-09-02-1009.excalidraw`).

[Untitled-2024-09-02-1009.excalidraw.json](https://github.com/user-attachments/files/17187353/Untitled-2024-09-02-1009.excalidraw.json)

**Comment by karlhorky** [comment]
> As a first step, can you uninstall the PWA and re-install it

> it worked fine repeatedly

Yeah, I did that before testing above. It worked at the first few tries, and I tricked myself that it was working properly.

But then after interacting more with the canvas state of Excalidraw (using "Reset the canvas"), then it stopped working.

Here's a repro with your file:


https://github.com/user-attachments/assets/6b8999f3-5104-43f6-8f25-8c790b71230b

**Comment by tomayac** [comment]
Hmm, at timestamp 0:37 the rectangle is visible, but then gets replaced. Tried the steps from your screencast locally, but still doesn't show up here. It looks more like some sort of race condition with the canvas reset, as the file was at least for a split second opened correctly.

**Comment by tomayac** [comment]
Can you try debugging locally with a `debugger` statement added right after `launchQueue.setConsumer`? It should be somewhere in `excalidraw/components/App.tsx` if I recall correctly.
